test/e2e/crud.go

```diff
@@ -44,49 +50,128 @@ var _ = Describe("Federated", func() {
 	// resources is performed in the event of test failure before
 	// controllers are shutdown.
 
+	validatedUnlabeledNotDeleted := false
 	for key := range typeConfigFixtures {
 		typeConfigName := key
 		fixture := typeConfigFixtures[key]
 		Describe(fmt.Sprintf("%q", typeConfigName), func() {
 			It("should be created, read, updated and deleted successfully", func() {
-				// Lookup the type config from the api
-				client, err := genericclient.New(f.KubeConfig())
-				if err != nil {
-					tl.Fatalf("Error initializing dynamic client: %v", err)
+				typeConfig, testObjectsFunc := getCrudTestInput(f, tl, typeConfigName, fixture)
+				crudTester, targetObject, overrides := initCrudTest(f, tl, typeConfig, testObjectsFunc)
+				crudTester.CheckLifecycle(targetObject, overrides)
+			})
+
+			// Unlabeled resource handling behavior should not vary between
+			// types, so testing a single type is sufficient.  Picking a
+			// namespaced type minimizes the impact of teardown failure.
+			if typeConfigName != "configmaps" || validatedUnlabeledNotDeleted {
+				return
+			}
+			validatedUnlabeledNotDeleted = true
+
+			It("should not be deleted if unlabeled", func() {
+				typeConfig, testObjectsFunc := getCrudTestInput(f, tl, typeConfigName, fixture)
+				crudTester, targetObject, _ := initCrudTest(f, tl, typeConfig, testObjectsFunc)
+
+				testClusters := crudTester.TestClusters()
+
+				By("Selecting a member cluster to create an unlabeled resource in")
+				clusterName := ""
+				for key := range testClusters {
+					if len(clusterName) == 0 {
```

Comment:
Did not understand the need of this. `clusterName := ""` is there two lines above.

Reply:
Done